tests/test_xlayer_cs.py

```diff
@@ -74,3 +78,45 @@ def test_PhysicalLink_setting_source_and_target(model: MelodyModel):
     assert source_pp == link.source
     assert target_pp == link.ends[1]
     assert target_pp == link.target
+
+
+def test_Component_parts(model: MelodyModel):
+    comp = model.by_uuid(HOGWARTS_UUID)
+
+    for part in comp.parts:
+        assert isinstance(part, cs.Part)
+
+
+@pytest.mark.parametrize(
+    "uuid",
+    [
+        pytest.param(HOGWARTS_UUID, id="Component"),
+        pytest.param(
+            "84c0978d-9a32-4f5b-8013-5b0b6adbfd73", id="ComponentPkg"
+        ),
+    ],
+)
+def test_component_creation_also_creates_a_part(model: MelodyModel, uuid: str):
+    name = "Test"
+    logical_parts = model.search("Part", below=model.la).by_name
+    assert name not in logical_parts, "Part already exists"  # type: ignore[operator]
+    obj = model.by_uuid(uuid)
+
+    comp = obj.components.create(name=name)
+
+    assert (part := model.search("Part", below=model.la).by_name(name))
+    assert isinstance(part, cs.Part)
+    assert part in comp.representing_parts
+    assert part.type == comp
+
+
+def test_component_modification_also_modifies_parts(model: MelodyModel):
+    name = "Test"
+
+    comp = model.by_uuid(HOGWARTS_UUID)
+    assert comp.representing_parts
+    comp.name = name
+    comp.allocated_functions.append(model.la.root_function)
+
+    assert (part := comp.representing_parts[0]).name == name
+    assert part.type.allocated_functions == comp.allocated_functions
```

Comment:
This assertion is wrong, what you really should be checking here is that `part.type == comp`. Or just remove it altogether, because that part of the model isn't touched in the test and therefore doesn't need checking.

```suggestion
```